### DISCUSSION TITLE
Support for EKS 1.21 upgrade

### DISCUSSION THREAD

**Discussion by pwoolcoc** [awslabs/aws-sdk-rust#532] [Q&A] [answered]
According to this page: https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-1.21 applications using other AWS sdks can upgrade their sdks to be compatible with the new service account token requirements. Is there support for this in the rust sdk or do we need to implement this token refreshing in our applications ourselves? If we need to do it ourselves, that's fine, I just want to make sure I'm not duplicating any work if all I need to do is upgrade the sdk to be compatible.

Thanks!

**Comment by Velfi** [comment]
@pwoolcoc How did you end up resolving this?

**Reply by pwoolcoc** [reply, marked as answer]
@Velfi I never did get a definitive answer about this, but we've been using `aws_config::web_identity_token::WebIdentityTokenCredentialsProvider` without problem for a while now.
